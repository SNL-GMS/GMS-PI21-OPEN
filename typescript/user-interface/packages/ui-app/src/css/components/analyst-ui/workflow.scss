@import '../../gms-fonts.scss';

$BLOCK_HEIGHT_PX: 24px;
$TABLE_BUTTON_WIDTH: 24px;
$TABLE_LABEL_WIDTH: 137px;
$WIDTH_OF_LEFT_CURTAIN: 24px;
$WIDTH_OF_RIGHT_CURTAIN: 140px;
$WIDTH_OF_RIGHT_PADDING: 148px;

.workflow-display-window {
  height: 100%;
  width: 100%;
  overflow: hidden;
  background-color: var(--gms-background, $gms-background);
}

.workflow-panel {
  height: 100%;
  width: 100%;
}

.workflow-table {
  display: flex;
  flex-direction: column;
}

.workflow-table-container {
  display: flex;
  position: relative;
  overflow-y: scroll;
  overflow-x: hidden;
  gap: 2px;
}

.workflow-table__curtain-left {
  width: $WIDTH_OF_LEFT_CURTAIN;
  position: absolute;
  background-color: var(--gms-background, $gms-background);
  z-index: 2;
}

.workflow-table__curtain-right {
  width: $WIDTH_OF_RIGHT_CURTAIN;
  position: absolute;
  right: 0px;
  background-color: var(--gms-background, $gms-background);
  z-index: 2;
  top: 0px;
}

.workflow-table__buttons {
  flex-direction: column;
  width: $TABLE_BUTTON_WIDTH;
  display: flex;
  align-items: flex-start;
}

.workflow-button {
  width: $TABLE_BUTTON_WIDTH;
}

.workflow-button__blank {
  height: $BLOCK_HEIGHT_PX;
  min-height: $BLOCK_HEIGHT_PX;
  z-index: 5;
  width: $TABLE_BUTTON_WIDTH;
}

.workflow-button__expand_collapse {
  height: $BLOCK_HEIGHT_PX;
  border-color: var(--gms-background, $gms-background);
  transition: margin-bottom 0.15s;
  z-index: 5;
  width: $TABLE_BUTTON_WIDTH;
}

.workflow-table__labels {
  display: flex;
  flex-direction: column;
}

.workflow-table-label {
  height: $BLOCK_HEIGHT_PX;
  width: $TABLE_LABEL_WIDTH;
  display: flex;
  justify-content: center;
  align-items: center;
  background: var(--gms-soft, $gms-soft);
  z-index: 5;
  -webkit-box-sizing: border-box;
  -moz-box-sizing: border-box;
  box-sizing: border-box;
  border: thin solid var(--gms-background, $gms-background);
}

.workflow-table-label--activity {
  filter: $interval-activity-darken-cell-by;
}

.workflow-table__time-axis {
  font-family: $gms-mono;
  margin-left: $TABLE_BUTTON_WIDTH;
  position: relative;
}

.time-axis-wrapper {
  width: 100%;
  height: 25px;
  flex: 0 0 auto;
  position: static;
  z-index: 1;
}

.scroll-bar-override {
  width: 100%;
  height: 14px;
  flex: 0 0 auto;
  position: static;
  z-index: 1;
}

.workflow-time-axis {
  path,
  line {
    fill: none;
    stroke: var(--gms-prominent, $gms-prominent);
    shape-rendering: crispEdges;
  }
  text {
    font-family: $gms-mono;
    font-size: 12px;
    fill: var(--gms-main, $gms-main);
    &.day-label {
      stroke: $workflow-day-divider;
      font-size: 13px;
    }
    &.first {
      text-anchor: start;
    }
    &.last {
      text-anchor: end;
    }
    &.hidden {
      display: none;
    }
  }
}

.time-axis {
  width: 100%;
  height: 25px;
  overflow: hidden;
}

.workflow-time-range {
  font-size: 0.75em;
  position: absolute;
  width: 100%;
  line-height: 1;
}

.time-range-left {
  float: left;
  line-height: 1;
}

.time-range-right {
  float: right;
  line-height: 1;
  margin-right: $WIDTH_OF_RIGHT_PADDING;
}

.stage-row__expand-button {
  height: $BLOCK_HEIGHT_PX;
  border: thin solid var(--gms-recessed, $gms-recessed);
  width: $BLOCK_HEIGHT_PX;
  transition: margin-bottom 0.15s;
  min-height: $BLOCK_HEIGHT_PX;
  min-width: $BLOCK_HEIGHT_PX;
  padding: 0;
  position: absolute;
  z-index: 5;
  -webkit-box-sizing: border-box;
  -moz-box-sizing: border-box;
  box-sizing: border-box;
}

.workflow-tooltip-container {
  display: grid;
  grid-template-columns: 2fr 3fr;
  margin: 0 auto;
  gap: 0.25em;
  padding: 0 0.5em;
  background-color: var(--gms-grid-background, $gms-grid-background);
}

.workflow-tooltip-container .label-value__value {
  color: var(--gms-main, $gms-main);
  font-size: 14px;
  &--numeric {
    font-family: $gms-mono;
  }
}

.workflow-tooltip-container .label-value__label {
  font-size: 14px;
}

.workflow-tooltip-container:last-child {
  padding-bottom: 0;
}

.workflow-tooltip-container + .workflow-tooltip-container {
  margin-top: 3px;
}

.interval-cell {
  display: flex;
  height: $BLOCK_HEIGHT_PX;
  align-items: center;
  text-align: center;
  justify-content: center;
  margin: 0;
  overflow: hidden;
  white-space: nowrap;
  .percent-bar {
    height: 99%;
    width: 99%;
  }
  -webkit-box-sizing: border-box;
  -moz-box-sizing: border-box;
  box-sizing: border-box;
  border: thin solid var(--gms-background, $gms-background);
}

.interval-cell:hover {
  filter: brightness(1.2);
}

.interval-cell--in-progress {
  background-color: $interval-in-progress-fill;
  .percent-bar {
    background-color: scale-color(
      $color: $interval-in-progress-fill,
      $lightness: 25%,
      $saturation: -15%
    );
  }
}

.interval-cell--complete {
  background-color: $interval-complete-fill;
}

.interval-cell--not-complete {
  background-color: $interval-not-complete-fill;
}

.interval-cell--not-started {
  background-color: $interval-not-started-fill;
}

.interval-cell--skipped {
  background-color: $interval-skipped-fill;
}

.interval-cell--failed {
  background-color: $interval-failed-fill;
}

.interval-cell--selected {
  border: 2px solid $gms-selection;
}

.interval-cell--activity-cell {
  filter: $interval-activity-darken-cell-by;
}

.interval-cell--activity-cell:hover {
  filter: brightness(1);
}

.interval-cell--stale {
  filter: brightness(0.5);
}

.interval-cell--clickable {
  cursor: pointer;
}

.workflow-day-boundary-container {
  position: absolute;
}

.workflow-day-divider {
  position: absolute;
  top: 0px;
  bottom: 0px;
  width: 2px;
  z-index: 1;
  margin-left: -1px;
  background-color: #ffc940;
}

.workflow-ellipsis {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  width: 100px;
  text-align: center;
}

.workflow-open-anything-row {
  display: flex;
  flex-flow: wrap;
  box-sizing: border-box;
  align-items: center;
  min-width: 56ch;
  & > * {
    margin-left: 0.5rem;
  }
}

.open-anything-contents .dropdown-container {
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-top: 7px;
}

.workflow-open-anything-row .dropdown-container .span {
  display: flex;
}

.open-anything-contents .dropdown-label {
  padding: 0 0.5em 0 0;
  line-height: 1;
  color: var(--core-prominent, var(--gms-prominent, $core-prominent));
  font-weight: 600;
  width: 35%;
}

.open-anything-contents .dropdown-selector {
  width: 65%;
}

.interval-confirmation-contents {
  display: flex;
}

.interval-confirmation-text {
  width: 90%;
}

.interval-confirmation-icon {
  width: 10%;
}

.interval-confirmation-header {
  font-weight: bold;
}

.open-anything-error {
  color: $gms-strong-warning;
  display: flex;
  min-height: 20px;
}
.open-anything-error-text {
  margin-left: 5px;
  width: calc(100% - 20px);
}
