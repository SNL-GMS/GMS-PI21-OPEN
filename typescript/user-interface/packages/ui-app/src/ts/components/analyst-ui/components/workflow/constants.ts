import { SECONDS_IN_MINUTES } from '@gms/common-util';

export const TABLE_SIDE_PADDING = 20;
export const TABLE_BUTTON_WIDTH = 32;
export const TABLE_LABEL_WIDTH = 135;
export const TOOLBAR_HEIGHT_PX = 35;
export const TIME_AXIS_HEIGHT = 35;

export const HUNDRED_TWENTY_SECOND_BUFFER = 120;
export const HEIGHT = 24;
export const PIXELS_PER_SECOND = 2 / SECONDS_IN_MINUTES;
