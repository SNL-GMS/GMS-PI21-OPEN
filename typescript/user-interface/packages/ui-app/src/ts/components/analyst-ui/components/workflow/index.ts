import { WorkflowComponent as Workflow } from './workflow-component';
import * as WorkflowUtil from './workflow-util';

export { Workflow, WorkflowUtil };
