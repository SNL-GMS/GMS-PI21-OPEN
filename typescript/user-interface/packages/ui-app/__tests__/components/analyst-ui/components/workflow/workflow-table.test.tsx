import type { StageIntervalList } from '@gms/ui-state';
import { waitForComponentToPaint } from '@gms/ui-state/__tests__/test-util';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import type Immutable from 'immutable';
import React from 'react';
import { act, create } from 'react-test-renderer';

import type { RowState } from '../../../../../src/ts/components/analyst-ui/components/workflow/workflow-table';
import { WorkflowTable } from '../../../../../src/ts/components/analyst-ui/components/workflow/workflow-table';
import { glContainer } from './gl-container';
import * as WorkflowDataTypes from './workflow-data-types';

window.ResizeObserver = jest.fn(() => {
  return { observe: jest.fn(), disconnect: jest.fn(), unobserve: jest.fn() };
});
const globalAny: any = global;
globalAny.ResizeObserver = window.ResizeObserver;
globalAny.DOMRect = jest.fn(() => ({}));

const intervalQueryResult: StageIntervalList = [];
intervalQueryResult.push({
  name: WorkflowDataTypes.interactiveStage.name,
  value: [WorkflowDataTypes.interactiveAnalysisStageInterval]
});

// eslint-disable-next-line @typescript-eslint/no-var-requires, @typescript-eslint/no-require-imports
const Enzyme = require('enzyme');

// set up window alert and open so we don't see errors
window.alert = jest.fn();
window.open = jest.fn();

const MOCK_TIME = 1606818240000;
global.Date.now = jest.fn(() => MOCK_TIME);

jest.spyOn(document, 'querySelector').mockImplementation(() => {
  return ({
    scrollWidth: 1200,
    clientWidth: 1200,
    getBoundingClientRect: jest.fn().mockReturnValue({
      width: 1200
    }),
    parentElement: {
      scrollTo: jest.fn()
    }
  } as unknown) as Element;
});
describe('Workflow Table', () => {
  it('is exported', () => {
    expect(WorkflowTable).toBeDefined();
  });

  it('matches snapshot', () => {
    const component = Enzyme.mount(
      <WorkflowTable
        glContainer={glContainer}
        timeRange={{
          startTimeSecs: 0,
          endTimeSecs: 360000
        }}
        widthPx={500}
        heightPx={500}
        stageIntervals={intervalQueryResult}
        workflow={WorkflowDataTypes.workflow}
        staleStartTime={123456}
      />
    );
    expect(
      toJson(component, {
        noKey: false,
        mode: 'deep'
      })
    ).toMatchSnapshot();

    component.setProps({
      timeRange: {
        startTimeSecs: 0,
        endTimeSecs: 360001
      },
      widthPx: 510,
      heightPx: 510,
      stageIntervals: intervalQueryResult,
      workflow: WorkflowDataTypes.workflow
    });

    const map: Immutable.Map<string, RowState> = component.state('expandedDataMap');
    const value: RowState = map.get(WorkflowDataTypes.interactiveStage.name);
    component.setState({
      expandedDataMap: map.set(WorkflowDataTypes.interactiveStage.name, {
        ...value,
        isExpanded: true
      })
    });

    expect(
      toJson(component, {
        noKey: false,
        mode: 'deep'
      })
    ).toMatchSnapshot();
  });
  describe('lifecycle methods', () => {
    it('mounts successfully if it finds virtualized container', () => {
      let result;
      // eslint-disable-next-line @typescript-eslint/no-floating-promises
      act(() => {
        result = create(
          <WorkflowTable
            glContainer={glContainer}
            timeRange={{
              startTimeSecs: 0,
              endTimeSecs: 360000
            }}
            widthPx={500}
            heightPx={500}
            stageIntervals={intervalQueryResult}
            workflow={WorkflowDataTypes.workflow}
            staleStartTime={123456}
          />
        );
      });
      expect(result.toJSON()).toMatchSnapshot();
    });
  });

  it('Horizontal scroll', async () => {
    const result = mount(
      <WorkflowTable
        glContainer={glContainer}
        timeRange={{
          startTimeSecs: 0,
          endTimeSecs: 360000
        }}
        widthPx={500}
        heightPx={500}
        stageIntervals={intervalQueryResult}
        workflow={WorkflowDataTypes.workflow}
        staleStartTime={123456}
      />
    );

    await waitForComponentToPaint(result);

    const spy = jest.spyOn(result.instance() as WorkflowTable, 'onWheel');
    const instance = result.instance() as WorkflowTable;

    // deltaX !== 0
    const wheelOne = {
      shiftKey: true,
      deltaX: 100,
      deltaY: 0
    };

    // deltaX === 0 and shiftKey === true
    const wheelTwo = {
      shiftKey: true,
      deltaX: 0,
      deltaY: 100
    };

    // deltaX === 0 and shiftKey === false
    const wheelThree = {
      shiftKey: false,
      deltaX: 0,
      deltaY: 100
    };

    instance.onWheel(wheelOne as React.WheelEvent<HTMLDivElement>);
    instance.onWheel(wheelTwo as React.WheelEvent<HTMLDivElement>);
    instance.onWheel(wheelThree as React.WheelEvent<HTMLDivElement>);
    expect(spy).toHaveBeenCalledTimes(3);
  });
});
