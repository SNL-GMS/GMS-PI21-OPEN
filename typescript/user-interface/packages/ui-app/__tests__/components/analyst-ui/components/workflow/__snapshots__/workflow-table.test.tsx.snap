// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Workflow Table lifecycle methods mounts successfully if it finds virtualized container 1`] = `
<div
  className="workflow-table"
  style={
    Object {
      "height": "430px",
    }
  }
>
  <div
    className="workflow-table-container"
  >
    <div
      className="workflow-table__curtain-left"
      style={
        Object {
          "height": "48px",
        }
      }
    />
    <div
      className="workflow-table__buttons"
    >
      <div
        className="workflow-button"
        data-cy="workflow-expand-AL1"
      >
        <div
          className="workflow-button__expand_collapse"
        >
          <button
            className="bp4-button stage-row__expand-button"
            onBlur={[Function]}
            onClick={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            type="button"
          >
            <span
              aria-hidden={true}
              className="bp4-icon bp4-icon-small-plus"
              icon="small-plus"
            >
              <svg
                data-icon="small-plus"
                height={16}
                viewBox="0 0 16 16"
                width={16}
              >
                <path
                  d="M11 7H9V5c0-.55-.45-1-1-1s-1 .45-1 1v2H5c-.55 0-1 .45-1 1s.45 1 1 1h2v2c0 .55.45 1 1 1s1-.45 1-1V9h2c.55 0 1-.45 1-1s-.45-1-1-1z"
                  fillRule="evenodd"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
      <div
        className="workflow-button"
        data-cy="workflow-expand-Auto Network"
      >
        <div
          className="workflow-button__expand_collapse"
        >
          <button
            className="bp4-button stage-row__expand-button"
            onBlur={[Function]}
            onClick={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            type="button"
          >
            <span
              aria-hidden={true}
              className="bp4-icon bp4-icon-small-plus"
              icon="small-plus"
            >
              <svg
                data-icon="small-plus"
                height={16}
                viewBox="0 0 16 16"
                width={16}
              >
                <path
                  d="M11 7H9V5c0-.55-.45-1-1-1s-1 .45-1 1v2H5c-.55 0-1 .45-1 1s.45 1 1 1h2v2c0 .55.45 1 1 1s1-.45 1-1V9h2c.55 0 1-.45 1-1s-.45-1-1-1z"
                  fillRule="evenodd"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
    <div>
      <div
        className="workflow-day-boundary-container"
        style={
          Object {
            "height": 48,
            "width": 12000,
          }
        }
      >
        <div
          className="workflow-day-divider"
          style={
            Object {
              "height": "100%",
              "left": "0px",
            }
          }
        />
        <div
          className="workflow-day-divider"
          style={
            Object {
              "height": "100%",
              "left": "2880px",
            }
          }
        />
        <div
          className="workflow-day-divider"
          style={
            Object {
              "height": "100%",
              "left": "5760px",
            }
          }
        />
        <div
          className="workflow-day-divider"
          style={
            Object {
              "height": "100%",
              "left": "8640px",
            }
          }
        />
        <div
          className="workflow-day-divider"
          style={
            Object {
              "height": "100%",
              "left": "11520px",
            }
          }
        />
      </div>
      <div
        data-cy="workflow-table__container"
        onMouseDown={[Function]}
        onMouseMove={[Function]}
        onMouseUp={[Function]}
        onWheel={[Function]}
        role="presentation"
        style={
          Object {
            "height": 48,
          }
        }
      >
        <div
          aria-label="grid"
          className="ReactVirtualized__Collection"
          onScroll={[Function]}
          role="grid"
          style={
            Object {
              "WebkitOverflowScrolling": "touch",
              "boxSizing": "border-box",
              "direction": "ltr",
              "height": 48,
              "overflowX": "auto",
              "overflowY": "hidden",
              "position": "relative",
              "width": 293,
              "willChange": "transform",
            }
          }
          tabIndex={0}
        >
          <div
            className="ReactVirtualized__Collection__innerScrollContainer"
            style={
              Object {
                "height": 48,
                "maxHeight": 48,
                "maxWidth": 54068572.9,
                "overflow": "hidden",
                "pointerEvents": "",
                "width": 54068572.9,
              }
            }
          >
            <div
              style={
                Object {
                  "height": 24,
                  "left": 0,
                  "position": "absolute",
                  "top": 0,
                  "width": 4,
                }
              }
            />
            <div
              style={
                Object {
                  "height": 24,
                  "left": 0,
                  "position": "absolute",
                  "top": 24,
                  "width": 4,
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="workflow-table__labels"
      style={
        Object {
          "height": "48px",
        }
      }
    >
      <div
        className="workflow-table-label"
        data-cy="workflow-table-label-AL1"
      >
        <div>
          AL1
        </div>
      </div>
      <div
        className="workflow-table-label"
        data-cy="workflow-table-label-Auto Network"
      >
        <div>
          Auto Network
        </div>
      </div>
    </div>
    <div
      className="workflow-table__curtain-right"
      style={
        Object {
          "height": "48px",
        }
      }
    />
  </div>
  <div
    className="workflow-table__time-axis"
  >
    <div
      className="scroll-bar-override"
      style={
        Object {
          "width": "293px",
        }
      }
    >
      <div
        className="scroll-bar scroll-box--x"
        data-cy="unknown-gutter-scrollbar-override"
        onClick={[Function]}
      >
        <div
          className="scroll-bar__bar scroll-box--x"
          data-cy="unknown-scrollbar-override"
          onClick={[Function]}
          onMouseDown={[Function]}
          style={
            Object {
              "transform": "translate(NaNpx, 0px)",
              "width": 0,
            }
          }
        />
      </div>
    </div>
    <div
      className="time-axis-wrapper"
      style={
        Object {
          "width": "293px",
        }
      }
    >
      <div
        className="time-axis"
      />
    </div>
    <div
      className="workflow-time-range"
    >
      <span
        className="time-range-left"
      >
        1970-01-01
      </span>
      <span
        className="time-range-right"
      >
        1970-01-05
      </span>
    </div>
  </div>
</div>
`;

exports[`Workflow Table matches snapshot 1`] = `
<div
  className="workflow-table"
  style={
    Object {
      "height": "430px",
    }
  }
>
  <div
    className="workflow-table-container"
  >
    <div
      className="workflow-table__curtain-left"
      style={
        Object {
          "height": "48px",
        }
      }
    />
    <div
      className="workflow-table__buttons"
    >
      <div
        className="workflow-button"
        data-cy="workflow-expand-AL1"
        key="expand AL1"
      >
        <div
          className="workflow-button__expand_collapse"
          key="main row AL1"
        >
          <Memo(StageExpansionButtonComponent)
            disabled={false}
            isExpanded={false}
            stageName="AL1"
            toggle={[Function]}
          >
            <button
              className="bp4-button stage-row__expand-button"
              onBlur={[Function]}
              onClick={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              type="button"
            >
              <span
                aria-hidden={true}
                className="bp4-icon bp4-icon-small-plus"
                icon="small-plus"
              >
                <svg
                  data-icon="small-plus"
                  height={16}
                  viewBox="0 0 16 16"
                  width={16}
                >
                  <path
                    d="M11 7H9V5c0-.55-.45-1-1-1s-1 .45-1 1v2H5c-.55 0-1 .45-1 1s.45 1 1 1h2v2c0 .55.45 1 1 1s1-.45 1-1V9h2c.55 0 1-.45 1-1s-.45-1-1-1z"
                    fillRule="evenodd"
                    key="0"
                  />
                </svg>
              </span>
            </button>
          </Memo(StageExpansionButtonComponent)>
        </div>
      </div>
      <div
        className="workflow-button"
        data-cy="workflow-expand-Auto Network"
        key="expand Auto Network"
      >
        <div
          className="workflow-button__expand_collapse"
          key="main row Auto Network"
        >
          <Memo(StageExpansionButtonComponent)
            disabled={false}
            isExpanded={false}
            stageName="Auto Network"
            toggle={[Function]}
          >
            <button
              className="bp4-button stage-row__expand-button"
              onBlur={[Function]}
              onClick={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              type="button"
            >
              <span
                aria-hidden={true}
                className="bp4-icon bp4-icon-small-plus"
                icon="small-plus"
              >
                <svg
                  data-icon="small-plus"
                  height={16}
                  viewBox="0 0 16 16"
                  width={16}
                >
                  <path
                    d="M11 7H9V5c0-.55-.45-1-1-1s-1 .45-1 1v2H5c-.55 0-1 .45-1 1s.45 1 1 1h2v2c0 .55.45 1 1 1s1-.45 1-1V9h2c.55 0 1-.45 1-1s-.45-1-1-1z"
                    fillRule="evenodd"
                    key="0"
                  />
                </svg>
              </span>
            </button>
          </Memo(StageExpansionButtonComponent)>
        </div>
      </div>
    </div>
    <div>
      <div
        className="workflow-day-boundary-container"
        style={
          Object {
            "height": 48,
            "width": 12000,
          }
        }
      >
        <div
          className="workflow-day-divider"
          key="0"
          style={
            Object {
              "height": "100%",
              "left": "0px",
            }
          }
        />
        <div
          className="workflow-day-divider"
          key="86400000"
          style={
            Object {
              "height": "100%",
              "left": "2880px",
            }
          }
        />
        <div
          className="workflow-day-divider"
          key="172800000"
          style={
            Object {
              "height": "100%",
              "left": "5760px",
            }
          }
        />
        <div
          className="workflow-day-divider"
          key="259200000"
          style={
            Object {
              "height": "100%",
              "left": "8640px",
            }
          }
        />
        <div
          className="workflow-day-divider"
          key="345600000"
          style={
            Object {
              "height": "100%",
              "left": "11520px",
            }
          }
        />
      </div>
      <div
        data-cy="workflow-table__container"
        onMouseDown={[Function]}
        onMouseMove={[Function]}
        onMouseUp={[Function]}
        onWheel={[Function]}
        role="presentation"
        style={
          Object {
            "height": 48,
          }
        }
      >
        <div
          aria-label="grid"
          className="ReactVirtualized__Collection"
          onScroll={[Function]}
          role="grid"
          style={
            Object {
              "WebkitOverflowScrolling": "touch",
              "boxSizing": "border-box",
              "direction": "ltr",
              "height": 48,
              "overflowX": "auto",
              "overflowY": "hidden",
              "position": "relative",
              "width": 293,
              "willChange": "transform",
            }
          }
          tabIndex={0}
        >
          <div
            className="ReactVirtualized__Collection__innerScrollContainer"
            style={
              Object {
                "height": 48,
                "maxHeight": 48,
                "maxWidth": 54068572.9,
                "overflow": "hidden",
                "pointerEvents": "",
                "width": 54068572.9,
              }
            }
          >
            <div
              key="0"
              style={
                Object {
                  "height": 24,
                  "left": 0,
                  "position": "absolute",
                  "top": 0,
                  "width": 4,
                }
              }
            />
            <div
              key="3"
              style={
                Object {
                  "height": 24,
                  "left": 0,
                  "position": "absolute",
                  "top": 24,
                  "width": 4,
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="workflow-table__labels"
      style={
        Object {
          "height": "48px",
        }
      }
    >
      <div
        className="workflow-table-label"
        data-cy="workflow-table-label-AL1"
        key="AL1"
      >
        <div>
          AL1
        </div>
      </div>
      <div
        className="workflow-table-label"
        data-cy="workflow-table-label-Auto Network"
        key="Auto Network"
      >
        <div>
          Auto Network
        </div>
      </div>
    </div>
    <div
      className="workflow-table__curtain-right"
      style={
        Object {
          "height": "48px",
        }
      }
    />
  </div>
  <div
    className="workflow-table__time-axis"
  >
    <div
      className="scroll-bar-override"
      style={
        Object {
          "width": "293px",
        }
      }
    >
      <div
        className="scroll-bar scroll-box--x"
        data-cy="unknown-gutter-scrollbar-override"
        onClick={[Function]}
      >
        <div
          className="scroll-bar__bar scroll-box--x"
          data-cy="unknown-scrollbar-override"
          onClick={[Function]}
          onMouseDown={[Function]}
          style={
            Object {
              "transform": "translate(NaNpx, 0px)",
              "width": 0,
            }
          }
        />
      </div>
    </div>
    <div
      className="time-axis-wrapper"
      style={
        Object {
          "width": "293px",
        }
      }
    >
      <div
        className="time-axis"
      />
    </div>
    <div
      className="workflow-time-range"
    >
      <span
        className="time-range-left"
      >
        1970-01-01
      </span>
      <span
        className="time-range-right"
      >
        1970-01-05
      </span>
    </div>
  </div>
</div>
`;

exports[`Workflow Table matches snapshot 2`] = `
<div
  className="workflow-table"
  style={
    Object {
      "height": "440px",
    }
  }
>
  <div
    className="workflow-table-container"
  >
    <div
      className="workflow-table__curtain-left"
      style={
        Object {
          "height": "48px",
        }
      }
    />
    <div
      className="workflow-table__buttons"
    >
      <div
        className="workflow-button"
        data-cy="workflow-expand-AL1"
        key="expand AL1"
      >
        <div
          className="workflow-button__expand_collapse"
          key="main row AL1"
        >
          <Memo(StageExpansionButtonComponent)
            disabled={false}
            isExpanded={false}
            stageName="AL1"
            toggle={[Function]}
          >
            <button
              className="bp4-button stage-row__expand-button"
              onBlur={[Function]}
              onClick={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              type="button"
            >
              <span
                aria-hidden={true}
                className="bp4-icon bp4-icon-small-plus"
                icon="small-plus"
              >
                <svg
                  data-icon="small-plus"
                  height={16}
                  viewBox="0 0 16 16"
                  width={16}
                >
                  <path
                    d="M11 7H9V5c0-.55-.45-1-1-1s-1 .45-1 1v2H5c-.55 0-1 .45-1 1s.45 1 1 1h2v2c0 .55.45 1 1 1s1-.45 1-1V9h2c.55 0 1-.45 1-1s-.45-1-1-1z"
                    fillRule="evenodd"
                    key="0"
                  />
                </svg>
              </span>
            </button>
          </Memo(StageExpansionButtonComponent)>
        </div>
      </div>
      <div
        className="workflow-button"
        data-cy="workflow-expand-Auto Network"
        key="expand Auto Network"
      >
        <div
          className="workflow-button__expand_collapse"
          key="main row Auto Network"
        >
          <Memo(StageExpansionButtonComponent)
            disabled={false}
            isExpanded={false}
            stageName="Auto Network"
            toggle={[Function]}
          >
            <button
              className="bp4-button stage-row__expand-button"
              onBlur={[Function]}
              onClick={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              type="button"
            >
              <span
                aria-hidden={true}
                className="bp4-icon bp4-icon-small-plus"
                icon="small-plus"
              >
                <svg
                  data-icon="small-plus"
                  height={16}
                  viewBox="0 0 16 16"
                  width={16}
                >
                  <path
                    d="M11 7H9V5c0-.55-.45-1-1-1s-1 .45-1 1v2H5c-.55 0-1 .45-1 1s.45 1 1 1h2v2c0 .55.45 1 1 1s1-.45 1-1V9h2c.55 0 1-.45 1-1s-.45-1-1-1z"
                    fillRule="evenodd"
                    key="0"
                  />
                </svg>
              </span>
            </button>
          </Memo(StageExpansionButtonComponent)>
        </div>
      </div>
    </div>
    <div>
      <div
        className="workflow-day-boundary-container"
        style={
          Object {
            "height": 48,
            "width": 12000.033333333333,
          }
        }
      >
        <div
          className="workflow-day-divider"
          key="0"
          style={
            Object {
              "height": "100%",
              "left": "0px",
            }
          }
        />
        <div
          className="workflow-day-divider"
          key="86400000"
          style={
            Object {
              "height": "100%",
              "left": "2879.9999999999995px",
            }
          }
        />
        <div
          className="workflow-day-divider"
          key="172800000"
          style={
            Object {
              "height": "100%",
              "left": "5759.999999999999px",
            }
          }
        />
        <div
          className="workflow-day-divider"
          key="259200000"
          style={
            Object {
              "height": "100%",
              "left": "8640px",
            }
          }
        />
        <div
          className="workflow-day-divider"
          key="345600000"
          style={
            Object {
              "height": "100%",
              "left": "11519.999999999998px",
            }
          }
        />
      </div>
      <div
        data-cy="workflow-table__container"
        onMouseDown={[Function]}
        onMouseMove={[Function]}
        onMouseUp={[Function]}
        onWheel={[Function]}
        role="presentation"
        style={
          Object {
            "height": 48,
          }
        }
      >
        <div
          aria-label="grid"
          className="ReactVirtualized__Collection"
          onScroll={[Function]}
          role="grid"
          style={
            Object {
              "WebkitOverflowScrolling": "touch",
              "boxSizing": "border-box",
              "direction": "ltr",
              "height": 48,
              "overflowX": "auto",
              "overflowY": "hidden",
              "position": "relative",
              "width": 303,
              "willChange": "transform",
            }
          }
          tabIndex={0}
        >
          <div
            className="ReactVirtualized__Collection__innerScrollContainer"
            style={
              Object {
                "height": 48,
                "maxHeight": 48,
                "maxWidth": 54068572.9,
                "overflow": "hidden",
                "pointerEvents": "",
                "width": 54068572.9,
              }
            }
          >
            <div
              key="0"
              style={
                Object {
                  "height": 24,
                  "left": 0,
                  "position": "absolute",
                  "top": 0,
                  "width": 4,
                }
              }
            />
            <div
              key="3"
              style={
                Object {
                  "height": 24,
                  "left": 0,
                  "position": "absolute",
                  "top": 24,
                  "width": 4,
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="workflow-table__labels"
      style={
        Object {
          "height": "48px",
        }
      }
    >
      <div
        className="workflow-table-label"
        data-cy="workflow-table-label-AL1"
        key="AL1"
      >
        <div>
          AL1
        </div>
      </div>
      <div
        className="workflow-table-label"
        data-cy="workflow-table-label-Auto Network"
        key="Auto Network"
      >
        <div>
          Auto Network
        </div>
      </div>
    </div>
    <div
      className="workflow-table__curtain-right"
      style={
        Object {
          "height": "48px",
        }
      }
    />
  </div>
  <div
    className="workflow-table__time-axis"
  >
    <div
      className="scroll-bar-override"
      style={
        Object {
          "width": "303px",
        }
      }
    >
      <div
        className="scroll-bar scroll-box--x"
        data-cy="unknown-gutter-scrollbar-override"
        onClick={[Function]}
      >
        <div
          className="scroll-bar__bar scroll-box--x"
          data-cy="unknown-scrollbar-override"
          onClick={[Function]}
          onMouseDown={[Function]}
          style={
            Object {
              "transform": "translate(NaNpx, 0px)",
              "width": 0,
            }
          }
        />
      </div>
    </div>
    <div
      className="time-axis-wrapper"
      style={
        Object {
          "width": "303px",
        }
      }
    >
      <div
        className="time-axis"
      />
    </div>
    <div
      className="workflow-time-range"
    >
      <span
        className="time-range-left"
      >
        1970-01-01
      </span>
      <span
        className="time-range-right"
      >
        1970-01-05
      </span>
    </div>
  </div>
</div>
`;
